src/lints/inherent_associated_pub_const_missing.ron

```diff
@@ -0,0 +1,64 @@
+SemverQuery(
+    id: "inherent_associated_pub_const_missing",
+    human_readable_name: "inherent impl's associated pub const removed",
+    description: "An inherent impl's associated public const removed or renamed",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#item-remove"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on ImplOwner {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        inherent_impl {
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            
+                            associated_constant {
+                                associated_constant: name @output @tag
+                                public_api_eligible @filter(op: "=", value: ["$true"])
```

Comment:
If you get a chance, it would be great to expand the test coverage to include checking that these `public_api_eligible` filters are applied properly.

Right now, I'm pretty sure we could remove those filters (lines 21 and 25) and the tests would still pass.

Test cases for this would include situations where `#[doc(hidden)]` exists on the `impl` block, on the `pub const` itself, or both. It could of course exist in both the baseline and the current versions, or exist only in one of them.

This lint should fire only when the constant is truly removed (i.e. using it produces a compilation error) and not when the constant stops being public API — as I'm sure you remember from writing a bunch of lints for that, that's going to be a different lint.